### PR TITLE
Add BCE loss

### LINKED ISSUE #872: BCELoss support

### LINKED ISSUE #873: BCEWithLogitsLoss support

### PR DESCRIPTION
**Pull request by pglusacTT**
Add support for BCE loss and BCE with logits loss

Closes #872, closes #873

### CODE REVIEW

File: forge/forge/op/loss.py [hunk 1]
```diff
@@ -119,3 +121,59 @@ def forward(self, predictions, labels):
...
+
+    @validate_shapes(min_dim=1, max_dim=2)
+    def forward(self, prediction, labels):
+        # BCE: -1 * (y * log(p) + (1 - y) * log(1 - p))
+        # First term: y * log(p)
+        log_prediction = Log("log", prediction)
+        first_term = Multiply("mul_lab_pred", labels, log_prediction)
+
+        one = Constant("one", constant=1.0)
+        one = align_shape(one, labels, "one")
```
Comment (ndrakulicTT) [inline forge/forge/op/loss.py:151]: Does Subtract dont support tensor - scalar as it was done in CrossEntropyLoss lines 80-81

**Review by pilkicTT — APPROVED** [review]

File: forge/forge/op/loss.py (hunk 1 above)
Comment (pglusacTT) [reply forge/forge/op/loss.py:151]: Subtract isn't the issue, it's the reduce operation afterwards that sometimes has issues with broadcasted operands

I think we should play it safe for now and manually broadcast constants, until that issue is resolved

File: forge/forge/op/loss.py [hunk 2]
```diff
@@ -119,3 +121,59 @@ def forward(self, predictions, labels):
...
+
+        # Second term: (1 - y) * log(1 - p)
+        one_minus_labels = Subtract("one_minus_labels", one, labels)
+        one_minus_prediction = Subtract("one_minus_prediction", one, prediction)
+        log_one_minus_prediction = Log("log_one_minus_prediction", one_minus_prediction)
+        second_term = Multiply("second_term", one_minus_labels, log_one_minus_prediction)
+
+        # -1 * (y * log(p) + (1 - y) * log(1 - p))
+        sum_terms = Add("sum_terms", first_term, second_term)
+        neg_one = Constant("neg_one", constant=-1.0)
```
[the remark this answers is not in the document]
Comment (pglusacTT) [reply forge/forge/op/loss.py:167]: Tried it, looks like the constant is implicitly converted to bfloat16
